### ISSUE TITLE
Batch normalization 2D

### ISSUE THREAD

**Issue by RadjaHachilif** [intel/mkl-dnn#367]
Hello people,
Can you please tell me how to use batch normalization for 2D call ( on a 4D input ) ?

Thank you

**Comment by emfomenk** [comment]
Sorry for delay.

Could you please elaborate on what do you mean *regular batch normalization call*?
Intel MKL-DNN performs regular batch normalization, so what kind of difference do you expect?

**Comment by RadjaHachilif** [comment]
I apologize, your example indeed is a 2D batch normalization over 4D inputs, I had a problem with the definition of 2d bn and i found this explanation in PyTorch documentation 

> Batch Normalization over a 4D input (a mini-batch of 2D inputs with additional channel dimension). The mean and standard-deviation are calculated per-dimension over the mini-batches .... Because the Batch Normalization is done over the `C` dimension, computing statistics on `(N, H, W)` slices, it's common terminology to call this Spatial Batch Normalization.

Thank you so much for your help.

**Comment by emfomenk** [comment]
Oh, I see.

Yes, Intel MKL-DNN's batch normalization does exactly what you found in PyTorch's documentation. In fact this algorithm should be pretty common across all major frameworks like TF, MxNet, PyTorch, and Caffe.

**Comment by emfomenk** [comment]
Close this for now.
Feel free to reopen or create a new issue if you have any further questions.
